packages/datetime/src/dateInput.tsx

```diff
@@ -316,7 +320,14 @@ export class DateInput extends AbstractComponent<IDateInputProps, IDateInputStat
     }
 
     private handleInputFocus = (e: React.FocusEvent<HTMLInputElement>) => {
-        const valueString = isMomentNull(this.state.value) ? "" : this.state.value.format(this.props.format);
+        let valueString: string;
+        if (isMomentNull(this.state.value)) {
+          valueString = "";
+        } else if (this.props.locale) {
+          valueString = this.props.locale ? this.state.value.locale(this.props.locale).format(this.props.format);
+        } else {
+          valueString = this.state.value.format(this.props.format);
```

Comment:
also lots of repetition here. maybe add a helper function that does the `if (locale) else` logic?